rust/hyperlane-core/src/metrics/agent.rs

```diff
@@ -0,0 +1,23 @@
+use std::time::Duration;
+
+use async_trait::async_trait;
+
+use crate::{ChainResult, U256};
+
+/// Interval for querying the prometheus metrics endpoint.
+/// This should be whatever the prometheus scrape interval is
+pub const METRICS_SCRAPE_INTERVAL: Duration = Duration::from_secs(60);
+
+/// Trait to be implemented by all chain-specific agent implementations,
+/// to support gathering agent metrics.
+#[async_trait]
+pub trait AgentMetricsFetcher: Send + Sync {
```

Comment:
Curious why a new trait is preferable instead of just putting it in `HyperlaneProvider` and using that? I see `get_balance` as a general piece of functionality that may be useful outside the context of agent metrics and I could see us expanding generic agent metrics to include other things that the provider can expose, like "latest block height for chain" etc.

I also tend to think that the abstractions for the chains should be closer to "this is a slim interface to interact with this chain that's mostly agnostic to _how_ it's used" when possible rather than "these abstractions are aware of how they're being used by agents". Similar deal for hyperlane-core, I think keeping it more agnostic to the details of what agents are is probably healthy

I see that this is a supertrait of HyperlaneProvider tho which does make sense